lib/index.js

```diff
@@ -808,6 +810,58 @@ GA.prototype.productClickedEnhanced = function(track) {
 };
 
 /**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListViewedEnhanced = function(track) {
+  var props = track.properties(); 
+  var products = track.products();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
+    window.ga('ec:addImpression', {
+      id: product.product_id,
+      category: track.category(),
+      list: props.list_id,
+      position: products.map(function(x) { return x.product_id; }).indexOf(product.product_id)
+    });
+  });
+  window.ga('send', 'pageview'); 
+  this.pushEnhancedEcommerce(track);
+};
+
+/**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListFilteredEnhanced = function(track) {
+  var props = track.properties();
+  var products = track.products();
+  var filters = props.filters.map(function(obj) { return obj.type + ':' + obj.value;}).join();
```

Comment:
This will error out if either filters or sorters aren't sent with an array. Checking for their existence first and, if they don't exist, either skipping this or defaulting them to an empty array would help avoid errors if the Track call lacks them. 👍